src/new-client/src/utils/__tests__/emitTooLong.test.ts

```diff
@@ -0,0 +1,35 @@
+import { emitTooLongMessage } from "utils/emitTooLong"
+import { testScheduler } from "utils/testScheduler"
+
+const TOO_LONG = { status: "too long" }
+
+const TAKING_TOO_LONG = 2_000
+
+const response = {
+  response: "I am a response",
+}
+
+describe("utils/emitTooLong", () => {
+  it("should emit normal response if response is faster than timer", () => {
+    testScheduler().run(({ expectObservable, cold }) => {
+      const input = cold("    ---(a|)", { a: response })
+      const expectedOutput = "---(b|)"
+      const result = input.pipe(emitTooLongMessage(TAKING_TOO_LONG, TOO_LONG))
+      expectObservable(result).toBe(expectedOutput, {
+        b: response,
+      })
+    })
+  })
+
+  it("should emit too long message if response is slowlier than timer", () => {
+    testScheduler().run(({ expectObservable, cold }) => {
+      const input = cold("    3s (a|)", { a: response })
+      const expectedOutput = "2s b 999ms (c|)"
```

Comment:
```suggestion
      const input = cold("    2s -(a|)", { a: response })
      const expectedOutput = "2s b(c|)"
```

This suggestion is testing the limit and also making sure that the marbles are better aligned, which makes it easier to understand what's happening in each moment.